prune/prune_test.go

```diff
@@ -0,0 +1,679 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prune
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	crFake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const namespace = "default"
+const app = "churro"
+
+var _ = Describe("Prune", func() {
+	var (
+		fakeClient   client.Client
+		fakeObj      client.Object
+		prunerConfig PrunerOption
+		podGVK       schema.GroupVersionKind
+		jobGVK       schema.GroupVersionKind
+	)
+	BeforeEach(func() {
+		testScheme, err := createSchemes()
+		Expect(err).Should(BeNil())
+
+		fakeClient = crFake.NewClientBuilder().WithScheme(testScheme).Build()
+		fakeObj = &corev1.Pod{}
+
+		// Create our function to configure our pruner
+		prunerConfig = func(p *Pruner) {
```

Comment:
Why using a new PrunerOption function rather than the existing withNamespace and withLabels?
My concern is that you can do that as the test is part of the prune package but library users would not be able to do the same as Pruner fields are not exposed.

Reply:
That's a good point. I made that before we had the existing `WithNamespace` and `WithLabels` options and it didn't cross my mind to change it. Thanks for bringing it up @fgiloux !